Sources/MapboxMaps/Foundation/MapView.swift

```diff
@@ -326,6 +331,64 @@ open class MapView: UIView {
             mapboxMap: mapboxMap)
     }
 
+    private func subscribeToLifecycleNotifications() {
+        if #available(iOS 13.0, *), bundle.infoDictionary?["UIApplicationSceneManifest"] != nil {
+            notificationCenter.addObserver(self,
+                                           selector: #selector(sceneWillEnterForeground(_:)),
+                                           name: UIScene.willEnterForegroundNotification,
+                                           object: window?.parentScene)
+            notificationCenter.addObserver(self,
+                                           selector: #selector(sceneDidEnterBackground(_:)),
+                                           name: UIScene.didEnterBackgroundNotification,
+                                           object: window?.parentScene)
+        } else {
+            notificationCenter.addObserver(self,
+                                           selector: #selector(appWillEnterForeground),
+                                           name: UIApplication.willEnterForegroundNotification,
+                                           object: nil)
+            notificationCenter.addObserver(self,
+                                           selector: #selector(appDidEnterBackground),
+                                           name: UIApplication.didEnterBackgroundNotification,
+                                           object: nil)
+        }
+    }
+
+    private func unsubscribeFromLifecycleNotifications() {
+        if #available(iOS 13.0, *) {
+            notificationCenter.removeObserver(self, name: UIScene.willEnterForegroundNotification, object: nil)
+            notificationCenter.removeObserver(self, name: UIScene.didEnterBackgroundNotification, object: nil)
+        }
+        notificationCenter.removeObserver(self, name: UIApplication.willEnterForegroundNotification, object: nil)
+        notificationCenter.removeObserver(self, name: UIApplication.didEnterBackgroundNotification, object: nil)
+    }
+
+    @objc private func appWillEnterForeground() {
+        displayLink?.isPaused = false
+    }
+
+    @objc private func appDidEnterBackground() {
+        displayLink?.isPaused = true
+    }
+
+    @available(iOS 13.0, *)
+    @objc private func sceneWillEnterForeground(_ notification: Notification) {
+        guard notification.object as? UIScene == window?.parentScene else { return }
```

Comment:
I don't think these checks are necessary since we only subscribe for a specific object.

Reply:
Removing them would allow you to share handlers with the app lifecycle notifications

Reply:
I added them here to protect against a situation when map view moves to a window without scene attached(there is no clear documentation on specifics of this). So map view will subscribe to lifecycle notifications with `nil` object. And this supposed to be a safeguard against it.

Reply:
So the scenario is, on iOS 13+ with scenes enabled, map view moves to a window that has a nil scene?

In that case, we would basically not get pausing and resuming until the window's scene became non-nil, right?

Not sure if this scenario is even possible.

Reply:
Yeah, in this scenario we won't get pausing and resuming, it either this or pausing and resuming by notifications from a random scene(if there would be no checks). But this scenario is definitely in the realm of possibility - windowScene is optional readwrite property.